packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -2606,13 +2606,15 @@ class _TimePickerDialogState extends State<TimePickerDialog> with RestorationMix
                 restorationId: 'time_picker_scroll_view_vertical',
                 child: AnimatedContainer(
                   width: allowedSize.width,
-                  height: allowedSize.height,
                   duration: _kDialogSizeAnimationDuration,
                   curve: Curves.easeIn,
+                  constraints: BoxConstraints(maxHeight: allowedSize.height),
```

Comment:
Apologies for letting this sit, I needed to find time to look into it. It still looks like it's expanding a bit too much when it doesn't need to.

What if we added a minHeight to this with the original minimum height `216`, then added a mainAxisAlignment of `spaceBetween` to the Column and removed the spacer and flex from the Flexible.

We want to: have a small space between the form and actions if possible. If the form expands, then we want the space to go away first until it's gone completely, _then_ start growing the dialog box.

Theoretically this combo would work. What do you think?

Reply:
Yup, I think this should work. Since we are doing spaceBetween so first that will shrink the space and then it will change dialog height.

Reply:
@MitchellGoodwin I have pushed the change, Just one change from your suggestion i.e. keeping it limited to input and inputOnly mode.